### ISSUE TITLE
Still maintained?

### ISSUE THREAD

**Issue by mathieumg** [rlidwka/yapm#29]
@rlidwka Is this repo still maintained? I see it hasn't been synced with upstream since last January.

We're still using yapm and while the improvements are great, it still tickles me that it's now falling behind the main npm. As great as they are, I could probably do without all the improvements yapm provides _but_ the JSON5 support, and that's solely for the comments support. I see https://github.com/npm/read-package-json/pull/46 which gives me hope, but I'm not holding my breath. 

Moreover, since the switch from npm/yapm 1.x to 2.x, the `--save` and `--save-dev` commands don't save back to the JSON5 anymore, even though the packages get installed. If this repo is still maintained I could always open a separate issue for that.

Thank you for your great work!

**Comment by rlidwka** [comment]
I was planning to update when npm@3.0 comes out (which would be a huge work anyway). But it isn't seem to be happening as soon as I would like.

If people are using it, then I should find some time to update it.

But you're right, an amount of updates needed is becoming a bit scary.

I wonder if there are other good package managers out there worth migrating to? Like [jspm](https://github.com/jspm/jspm-cli)? 

Or maybe return to good old `yapm@0.x` that wasn't a fork of npm, but it a wrapper for `fs.*` calls. It's much easier to maintain.

**Comment by mathieumg** [comment]
Thanks for the quick reply! Yeah, as much as I'd like to believe npm3 is "right around the corner", I'm pretty sure it could still be a little while before we get to use it.

I'm not sure if I'd be ready to migrate to another package manager, out of fear of the unknown perhaps... What was the feature set of `yapm@0.x`? If it's just wrapping for JSON5 support around the current version of npm, that would already by huge. Thanks!

**Comment by rlidwka** [comment]
> What was the feature set of yapm@0.x? If it's just wrapping for JSON5 support around the current version of npm, that would already by huge. Thanks!

Yes. Here it is for historic purposes:
https://github.com/rlidwka/yapm/tree/good-old-wrapper

I'm not sure it'll even work right now.

PS: is there any crucial changes in npm functionality recently? 'cause I was under impression that they are mostly refactoring stuff.

**Comment by mathieumg** [comment]
Thanks!

> is there any crucial changes in npm functionality recently?

I do not feel like there is, but some usability enhancements are welcome fixes.

**Comment by mathieumg** [comment]
Yeah, as expected (I'm sure you aren't surprised either) npm would have nothing of the PR I linked to above.

**Comment by mathieumg** [comment]
@rlidwka 

> is there any crucial changes in npm functionality recently?

I'm sorry, this just came back to mind... Scoped packages is something fairly new we would like to leverage: https://docs.npmjs.com/misc/scope

**Comment by rlidwka** [comment]
> Scoped packages is something fairly new we would like to leverage

That's last year news, it was implemented in npm around August 2014.

**Comment by mathieumg** [comment]
Really? Well, my bad, I was sure I had read they were from circa April 2015. Perhaps that's just when they launched the npm private packages to go with it? (We use your sinopia module anyway! :smile:)

**Comment by mathieumg** [comment]
> I was planning to update when npm@3.0 comes out

Pre-release now up! https://github.com/npm/npm/releases/tag/v3.0.0

Is there anything I can do to help in that regard? Thanks.

**Comment by mathieumg** [comment]
> Yes. Here it is for historic purposes:
> https://github.com/rlidwka/yapm/tree/good-old-wrapper
> 
> I'm not sure it'll even work right now.

Just tested it and it doesn't, sadly. :( I'll try to experiment further with that and the npm3 beta.

**Comment by mathieumg** [comment]
npm3 is now out of beta, fwiw.

**Comment by wiredmax** [comment]
For anyone interested, @mathieumg and I have published [npm-json5](https://github.com/wiredmax/npm-json5) and it now works without a hitch with the latest version of npm3.
